### ISSUE TITLE
[Tooling] Add a `tox` environment for `black`

### ISSUE THREAD

**Issue by kdestin** [Azure/azure-sdk-for-python#28784]
**Is your feature request related to a problem? Please describe.**

Quality checks (formatting, linting, etc...) for packages in this repo are enforced by required PR checks. But waiting on a PR check can be a large time sink to get feedback that you forgot to format your code, fix a small lint, etc...

There is not currently a standardized way to invoke our code formatting tool `black` so that it uses the same version and config as the one that runs in our CI.

**Describe the solution you'd like**

1. Add a `tox` environment for `black`
2. Allow `scripts/devops_tasks/validate_formatting.py` to forward parameters to `black` (allows it to run on a subset of files)
3. Allow `scripts/devops_tasks/validate_formatting.py` to pass output through to the terminal (this would be implemented in a way that doesn't break existing workflows)
4. ~~(Nice to have) Move the `black` configuration to a separate file, and reference that config file with `black -c path/to/config` in `scripts/devops_tasks/validate_formatting.py`~~ Resolved by #28835 
5. (Nice to have) Rename `scripts/devops_tasks/validate_formatting.py` to something that makes it clear that it modifies files on disk.

**Describe alternatives you've considered**

N/A

**Additional context**

N/A

cc: @scbedd

**Comment by scbedd** [comment]
RE point 5:

We should definitely just rename it to `run_black.py`. That's all it's doing anyway!
